packages/visual-editor/src/components/Dropzone/useComponentProps.ts

```diff
@@ -88,6 +89,32 @@ export const useComponentProps = ({
             [variableName]: designValue,
           };
         } else if (variableMapping.type === 'BoundValue') {
+          if (!areEntitiesFetched) {
+            console.warn(
+              `[exp-builder.sdk::Dropzone::useComponentProps::'BoundValue'] Idle-cycle: as entities are not fetched(areEntitiesFetched=${areEntitiesFetched}), we cannot resolve bound values for ${variableName} so we just resolve them to default values.`
```

Comment:
Great comment Sofia! 🙇  

The reason that I added this if-statement checking explicitly for the idle-lifecycle, is because this logic was there "implicitly". 

if you check L103 on the left diff of this file, you will see `areEntitiesFetched` is checked and some values resolve to `undefined` and in very hidden implicit way we handle this idle-lifecycle. 
I just refactored this idle-lifecycle handling into an explicit case so that it's not hidden. 
Let me know what you think? 

ps. as to console.warn i changed it to console.debug (which we use a lot around SDK)

```
 if (!boundValue) {
            const maybeBoundAsset = areEntitiesFetched
              ? entityStore.getValue(binding, path.slice(0, -2))
              : undefined;
            if (isLinkToAsset(maybeBoundAsset)) {
              boundValue = maybeBoundAsset;
            }
          }
```

Reply:
And the reason I wanted it to track it with console.log is that from my understanding of this logic, it may hypothetically cause "flash of default values" (in case filling in entities takes longer time).

I haven't seen such a "flash of default values" yet, but from the code it seems it should happen 🤔

Reply:
I converted it `console.warn -> console.debug` and made prefix consistent with the rest of the SDK log messages

![image](https://github.com/contentful/experience-builder/assets/113918657/84854f44-acb3-4a05-bf70-397c1df69c21)